medicines/doc-index-updater/src/delete_manager/mod.rs

```diff
@@ -18,22 +18,28 @@ pub async fn delete_service_worker(
         tracing::error!("{:?}", e);
         anyhow!("Couldn't create the delete queue client")
     })?;
+    let search_client = search_client::factory();
+    let storage_client = storage_client::factory().map_err(|e| {
+        tracing::error!("{:?}", e);
+        anyhow!("Couldn't create storage client")
+    })?;
 
     loop {
         let message_result: Result<DeleteMessage, RetrieveFromQueueError> =
             delete_client.receive().await;
         match message_result {
             Ok(message) => {
                 tracing::info!("{:?} message receive!", message);
-
-                let blob_name = get_blob_name_from_content_id(&message.document_content_id).await?;
-                delete_blob(&storage_container_name, &blob_name)
+                let blob_name =
+                    get_blob_name_from_content_id(&search_client, message.document_content_id)
+                        .await?;
+                delete_from_index(&search_client, &blob_name).await?;
```

Comment:
is there a risk of the `search_client` failing and being unrecoverable?

Reply:
I don't think so. `search_client` is effectively a container for a bunch of configuration and a `reqwest` client… I'll read up on `reqwest` to check though.

Reply:
So long as it's making a new connection each time, it'll be fine. Only thought of it because I saw the factory moved out of the loop and we previously did something similar with a redis client, and had to be sure the connections themselves weren't being created as some kind of singleton.